File: leetcode/typescript/[724]Find-Pivot-Index.ts
// ピボットインデックスの抽出
function pivotIndex(nums: number[]): number {
  const totalSum: number = nums.reduce((a, b) => {
    return a + b;
  });

  let leftSum: number = 0;

  for (let i = 0; i < nums.length; i++) {
    if (totalSum - leftSum - nums[i] === leftSum) {
      return i;
    }
    leftSum += nums[i];
  }
  return -1;
}

File: leetcode/python/[724]Find-Pivot-Index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        totalSum = 0
        for i in nums:
            totalSum += i
        leftSum = 0
        for idx,item in enumerate(nums):
            if totalSum - leftSum - nums[idx] == leftSum:
                return idx
            leftSum += nums[idx]
        return -1
